municipal_finance/summarise_data.py

```diff
@@ -0,0 +1,119 @@
+from django.db.models import Min, Max
+from django_q.tasks import async_task
+from django.db import transaction
+
+from municipal_finance.models.data_summaries import Summary
+
+from scorecard.models.geography import Geography
+
+from municipal_finance.models.cash_flow import (
+    CflowFactsV1,
+    CflowFactsV2,
+)
+from municipal_finance.models.income_expenditure import (
+    IncexpFactsV1,
+    IncexpFactsV2,
+)
+from municipal_finance.models.financial_position import (
+    BsheetFactsV1,
+    FinancialPositionFactsV2,
+)
+from municipal_finance.models.capital import (
+    CapitalFactsV1,
+    CapitalFactsV2,
+)
+from municipal_finance.models.grants import (
+    ConditionalGrantFactsV1,
+    GrantFactsV2,
+)
+from municipal_finance.models.repairs_maintenance import (
+    RepairsMaintenanceFactsV1,
+    RepairsMaintenanceFactsV2,
+)
+from municipal_finance.models.aged_debtor import (
+    AgedDebtorFactsV1,
+    AgedDebtorFactsV2,
+)
+from municipal_finance.models.aged_creditor import (
+    AgedCreditorFactsV1,
+    AgedCreditorFactsV2,
+)
+from municipal_finance.models.uifw_expense import UIFWExpenseFacts
+
+
+FACT_TABLES = [
+    CflowFactsV1,
+    CflowFactsV2,
+    IncexpFactsV1,
+    IncexpFactsV2,
+    BsheetFactsV1,
+    FinancialPositionFactsV2,
+    CapitalFactsV1,
+    CapitalFactsV2,
+    ConditionalGrantFactsV1,
+    GrantFactsV2,
+    RepairsMaintenanceFactsV1,
+    RepairsMaintenanceFactsV2,
+    AgedDebtorFactsV1,
+    AgedDebtorFactsV2,
+    AgedCreditorFactsV1,
+    AgedCreditorFactsV2,
+    UIFWExpenseFacts
+]
+
+
+def summarise_task(task):
+    if task.success:
+        async_task(
+            "municipal_finance.summarise_data.summarise",
+            task_name="Summarise Data",
+        )
+
+
+@transaction.atomic
+def summarise():
+    min_year = 3000
+    max_year = 1000
+    count_facts = 0
+
+    for table in FACT_TABLES:
+        min_year = get_min(min_year, table)
+        max_year = get_max(max_year, table)
+
+    count_years = max_year - min_year
```

Comment:
is this an off by one error? looks like when I upload a file with a single year of data, it says I have zero years of data

Reply:
Changed to count a list of years and get min max from that without magic numbers